net-dns/dnrd: EAPI7 revbump, improve ebuild

mm1ke opened the pull request:
Hi,

This PR/Bug updates net-dns/dnrd for EAPI7 with some minor improvements to the ebuild and the init script.
Please review.

```diff
--- dnrd-2.20.3-r1.ebuild       2018-09-30 21:33:03.356550840 +0200
+++ dnrd-2.20.3-r2.ebuild       2018-10-03 19:39:00.214035275 +0200
@@ -1,38 +1,35 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2018 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=0
+EAPI=7
 
-inherit autotools eutils user
+inherit autotools user
 
 DESCRIPTION="A caching DNS proxy server"
 HOMEPAGE="http://dnrd.sourceforge.net/"
 SRC_URI="mirror://sourceforge/dnrd/${P}.tar.gz"
+
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~ppc ~x86"
 IUSE="debug"
-DEPEND=""
 
-src_unpack() {
-       unpack ${A}
-       cd "${S}"
-       epatch "${FILESDIR}"/${P}-docdir.patch
+PATCHES=( "${FILESDIR}"/${P}-docdir.patch )
+
+src_prepare() {
+       default
        eautoreconf
 }
 
-src_compile() {
+src_configure() {
        econf \
                $(use_enable debug) \
                --disable-dependency-tracking \
-               --docdir=/usr/share/doc/${PF} \
-               || die
-
-       emake || die
+               --docdir=/usr/share/doc/${PF}
 }
 
 src_install() {
-       emake DESTDIR="${D}" install || die
+       default
 
        keepdir /etc/dnrd
        doinitd ${FILESDIR}/dnrd
```

Closes: https://bugs.gentoo.org/667660
Signed-off-by: Michael Mair-Keimberger <m.mairkeimberger@gmail.com>

mgorny commented on net-dns/dnrd/dnrd-2.20.3-r2.ebuild:
```diff
@@ -0,0 +1,42 @@
+HOMEPAGE="http://dnrd.sourceforge.net/"
+SRC_URI="mirror://sourceforge/dnrd/${P}.tar.gz"
+
+LICENSE="GPL-2"
```
Verified that?

mm1ke replied:
Yes, it's correct

mgorny replied:
is it?

```
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
```

mgorny commented on net-dns/dnrd/dnrd-2.20.3-r2.ebuild:
```diff
@@ -0,0 +1,42 @@
+src_configure() {
+	econf \
+		$(use_enable debug) \
+		--disable-dependency-tracking \
```
Aren't those two passes as part of EAPI?

mgorny commented on net-dns/dnrd/dnrd-2.20.3-r2.ebuild:
```diff
@@ -0,0 +1,42 @@
+
+src_configure() {
+	econf \
+		$(use_enable debug) \
```
What does this do?

mgorny commented on net-dns/dnrd/files/dnrd:
```diff
@@ -11,13 +11,13 @@ depend() {
 start() {
 	ebegin "Starting dnrd"
-	/usr/sbin/dnrd $DNRD_OPTS &> /dev/null &
+	/usr/sbin/dnrd $DNRD_OPTS >/dev/null 2>&1
```
You're losing `&`, so I'm going to guess it's going to block forever? Also use start-stop-daemon maybe.

mm1ke commented:
@mgorny 
```--disable-dependency-tracking``` is being passed by the EAPI so i've removed it.. (actually didn't noticed that yet..), however as fas i saw debug isn't passed by any EAPI. Anyway, since it won't build with ```debug``` enabled i've removed it (also doesn't build with the r1). Don't even know what exactly it does.
Also updated the init script to use ```start-stop-daemon```

mm1ke commented:
Updated this PR once again and removed ```src_configure``` completely. ```docdir``` is also passed by EAPI7 so we don't need to set it :)

mgorny commented on net-dns/dnrd/dnrd-2.20.3-r2.ebuild:
```diff
@@ -0,0 +1,34 @@
+	default
+
+	keepdir /etc/dnrd
+	doinitd ${FILESDIR}/dnrd
```
Repoman doesn't like you:

```
   net-dns/dnrd/dnrd-2.20.3-r2.ebuild: Unquoted Variable on line: 27
   net-dns/dnrd/dnrd-2.20.3-r2.ebuild: Unquoted Variable on line: 28

```

mm1ke replied:
Ups, sorry, fixed :)
